src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/StatefulMarshallingStrategy.cs

```diff
@@ -407,6 +391,17 @@ public IEnumerable<StatementSyntax> GenerateCleanupStatements(TypePositionInfo i
 
         public IEnumerable<StatementSyntax> GenerateMarshalStatements(TypePositionInfo info, StubCodeContext context)
         {
+            if (context.Direction == MarshalDirection.ManagedToUnmanaged && !info.IsByRef && info.ByValueContentsMarshalKind == ByValueContentsMarshalKind.Out)
+            {
+                yield return _elementsMarshalling.GenerateManagedToUnmanagedByValueOutMarshalStatement(info, context);
+                yield break;
+            }
+            if (context.Direction == MarshalDirection.UnmanagedToManaged && !info.IsByRef && info.ByValueContentsMarshalKind.HasFlag(ByValueContentsMarshalKind.Out))
+            {
+                yield return _elementsMarshalling.GenerateUnmanagedToManagedByValueOutMarshalStatement(info, context);
+                yield break;
+            }
```

Comment:
Why for ManagedToUnmanaged to we check for `== Out` and for UnmanagedToManaged we check for `HasFlag(Out)`?

Reply:
I'll add some comments here. The `== Out` cases are checking "Has Out and not In". In these cases, we just want to clear the "source" collection as the passed in values are unused. Whereas when we do `HasFlag(Out)`, we want to get both `[Out]` and `[In, Out]` cases since that's the actual out marshalling scenario.